### PR TITLE
fix broken link the readme and few minor edits

### PR DESCRIPTION
**Pull request by msherif1234**
few minor edits to readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ xdp-pass-all-nodes   pass              {}             0          {"primarynodein
 
 ## API Types Overview
 
-Refer to [api-spec.md](./api-spec.md) for a detailed description of all the bpfman Kubernetes API types.
+Refer to [api-spec.md](https://github.com/bpfman/bpfman/blob/main/docs/developer-guide/api-spec.md) for a detailed description of all the bpfman Kubernetes API types.
```
Comment (Billy99) [inline README.md:151]: I think this is the link you want: https://bpfman.io/main/developer-guide/api-spec/
When I clicked on the link you provided, this was the content:
```
API Specification
--8<-- "./bpfman-operator/apidocs.html"
```

**Review by Billy99 — COMMENTED** [review]
Thanks, just one comment

**Review by anfredette — APPROVED** [review]
lgtm

**Review by anfredette — DISMISSED** [review]
/lgtm

**Review by astoycos — DISMISSED** [review]
LGTM

Thanks @msherif1234!

**Review by Billy99 — DISMISSED** [review]
/lgtm

**Review by anfredette — APPROVED** [review]
LGTM

**Pull request merged**